codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -164,4 +183,51 @@ class FileCodesignVisitor {
       processManager: processManager,
     );
   }
+
+  /// Visit and codesign a binary with / without entilement.
+  ///
+  /// At this stage, the virtual [entitlementCurrentPath] accumulated through the recursive visit, is compared
+  /// against the paths extracted from [fileWithEntitlements], to help determine if this file should be signed
+  /// with entitlements.
+  Future<void> visitBinaryFile({required FileSystemEntity binaryEntity, required String entitlementParentPath}) async {
```

Comment:
I'm thinking `binaryEntity` can be a `File` rather than `FileSystemEntity`, right? That is to say, we should never be passing a directory to this method.

Reply:
Yeah `visitBinary` is reached if the file type is `application/x-mach-binary`, but for sure I have changed it to be a `File` type.